cmd/daemonset-check/main.go

```
@@ -21,1118 +6,217 @@ import (
-		// The check has timed out because its time to run again
-		dsc.cancelFunc() // cancel context
-		errorMessage := "Failed to complete checks for " + dsc.Name() + " in time!  Next run came up but check was still running."
-		//log.Errorln(dsc.Name(), errorMessage)
-		err = checkclient.ReportFailure([]string{errorMessage})
-		if err != nil {
-			log.Println("Error reporting failure to Kuberhealthy servers:", err)
-			return err
-		}
-		log.Println("Successfully reported failure to Kuberhealthy servers")
-	case <-time.After(dsc.Timeout()):
-		// The check has timed out after its specified timeout period
-		dsc.cancelFunc() // cancel context
-		errorMessage := "Failed to complete checks for " + dsc.Name() + " in time!  Timeout was reached."
-		//log.Errorln(dsc.Name(), errorMessage)
-		err = checkclient.ReportFailure([]string{errorMessage})
-		if err != nil {
-			log.Println("Error reporting failure to Kuberhealthy servers:", err)
-			return err
-		}
-		log.Println("Successfully reported failure to Kuberhealthy servers")
-	case err := <-doneChan:
-		dsc.cancelFunc()
-		if err != nil {
-			log.Println("Check failed with error:", err)
-			err = checkclient.ReportFailure([]string{err.Error()})
-			if err != nil {
-				log.Println("Failed to report error to Kuberhealthy servers:", err)
-				return err
-			}
-			log.Println("Successfully reported error to Kuberhealthy servers")
-			return nil
-		}
-		err = checkclient.ReportSuccess()
+	case err = <-runCheckDoneChan:
 		if err != nil {
-			log.Println("Error reporting success to Kuberhealthy servers:", err)
-			return err
+			reportErrorsToKuberhealthy([]string{"kuberhealthy/daemonset: " + err.Error()})
+		} else {
+			reportOKToKuberhealthy()
 		}
-		log.Println("Successfully reported success to Kuberhealthy servers")
+		log.Infoln("Done running daemonset check")
+	case <-signalChan:
+		log.Infoln("Received shutdown signal. Canceling context and proceeding directly to cleanup.")
+		reportOKToKuberhealthy()
```

added a `reportOKToKuberhealthy` since if no report is made, kuberhealthy defaults the error to: `Check execution error: kuberhealthy/daemonset: timed out waiting for checker pod to report in` . not sure if we want to report an error instead